source/sn_coap_protocol.c

```diff
@@ -1472,27 +1466,28 @@ static void sn_coap_protocol_handle_blockwise_timout(struct coap_s *handle)
     ns_list_foreach_safe(coap_blockwise_msg_s, removed_blocwise_msg_ptr, &handle->linked_list_blockwise_sent_msgs) {
         if ((handle->system_time - removed_blocwise_msg_ptr->timestamp)  > SN_COAP_BLOCKWISE_MAX_TIME_DATA_STORED) {
 
+            // Item must be removed from the list before calling the rx_callback function
+            // since the callback may access the same data
```

Comment:
This is slightly misleading, as the real problem was that the callback could actually clear the list and therefore free the item already and cause a use after free after callback returns.